mod-source-record-manager-server/src/main/java/org/folio/services/MappingRuleCache.java

```diff
@@ -0,0 +1,65 @@
+package org.folio.services;
+
+import com.github.benmanes.caffeine.cache.AsyncLoadingCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.folio.dao.MappingRuleDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+@Component
+public class MappingRuleCache {
+
+  private static final Logger LOG = LoggerFactory.getLogger(MappingRuleCache.class);
+
+  private MappingRuleDao mappingRuleDao;
+  private AsyncLoadingCache<String, Optional<JsonObject>> cache;
+
+  @Autowired
+  public MappingRuleCache(MappingRuleDao mappingRuleDao, Vertx vertx) {
+    this.mappingRuleDao = mappingRuleDao;
+    cache = Caffeine.newBuilder()
+      .executor(task -> vertx.runOnContext(ar -> task.run()))
+      .maximumSize(1)
```

Comment:
I think it would be better to extract this number(1) to the property(field in this class). For better reading and fixing some issues in the future.

Reply:
missed it, it should be deleted since cache can stores rules for multiple tenants, thanks